python/lsst/verify/compatibility/metricTask.py

```diff
@@ -0,0 +1,196 @@
+# This file is part of verify.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import abc
+
+import lsst.pex.config
+import lsst.pipe.base as pipeBase
+
+__all__ = ["MetricTask"]
+
+
+class MetricTask(pipeBase.Task, metaclass=abc.ABCMeta):
+    """A ``MetricTask`` object computes exactly one metric, based on
+    arbitrary input datasets.
+
+    Both the metric and the input data are configurable. Metrics may be
+    associated with a data ID at any level of granularity, including
+    repository-wide. MetricTasks should assume they take multiple input
+    datasets, regardless of the expected metric granularity, unless it is
+    impossible for more than one copy of a dataset to exist.
+
+    All input data shall be considered optional in order to maximize the
+    metric's usefulness for incomplete pipeline runs or runs with optional
+    processing steps. If a metric cannot be calculated because the necessary
+    inputs are missing, the ``MetricTask`` must return `None` in place of
+    the measurement.
+
+    .. note::
+        The API is designed to make it easy to convert all ``MetricTasks`` to
+        `~lsst.pipe.base.PipelineTask` later, but this class is *not* a
+        `~lsst.pipe.base.PipelineTask` and will not work with activators,
+        quanta, or `lsst.daf.butler`.
+
+    Parameters
+    ----------
+    Same as for `lsst.pipe.base.PipelineTask`.
+    """
+
+    # There may be a specialized MetricTaskConfig later, details TBD
+    ConfigClass = lsst.pex.config.Config
+
+    def adaptArgsAndRun(self, inputData, inputDataIds, outputDataId):
+        """Compute a metric from in-memory data.
+
+        This implementation calls `run` on the contents of ``inputData``,
```

Comment:
No. Perhaps the word "implementation" threw you off (I'm borrowing from the Java convention, since AFAIK Python doesn't care much for inheritance safety), but this is a *behavioral* description of `adaptArgsAndRun`, specifically how its behavior will be affected if a subclass overrides `run`.